framework/common/encryption.cpp

```diff
@@ -134,6 +135,14 @@ bool diffie_hellman::compute_key(uint8_t *key, unsigned &key_length, const uint8
     return true;
 }
 
+void diffie_hellman::copy_pubkey(uint8_t *dest) const
+{
+    int padding = max_pubkey_length - m_pubkey_length;
+    mapf_assert(padding >= 0);
```

Comment:
This is not really doing anything in release builds, I think it is more appropriate to at least add a LOG(FATAL) here.

Reply:
That should be part of mapf_assert then.